test/unit/lib/migration-payloads/validation/payload-validation-id-change.spec.js

```diff
@@ -0,0 +1,225 @@
+'use strict';
+
+const { expect } = require('chai');
+const Bluebird = require('bluebird');
+
+const migrationPayloads = require('../../../../../lib/migration-payloads');
+const migrationPlan = require('../../../../../lib/migration-plan');
+const migrationChunks = require('../../../../../lib/migration-chunks');
+const migrationSteps = require('../../../../../lib/migration-steps');
+const validatePayloads = require('../../../../../lib/migration-payloads/validation');
+
+describe('payload validation (dependencies)', function () {
+  describe('when setting a new id but it is the same as the old', function () {
+    it('returns an error', Bluebird.coroutine(function * () {
+      const steps = yield migrationSteps(function (migration) {
+        migration
+          .editContentType('book')
+          .editField('title')
+          .id('title');
+      });
+
+      const existingCts = [
+        {
+          sys: {
+            version: 1,
+            id: 'book'
+          },
+          name: 'Book',
+          fields: [{
+            id: 'title',
+            name: 'Title',
+            type: 'Symbol'
+          }]
+        }
+      ];
+
+      const chunks = migrationChunks(steps);
+      const plan = migrationPlan(chunks);
+      const payloads = migrationPayloads(plan, existingCts);
+      const errors = validatePayloads(payloads);
```

Comment:
Just a random thought: to not always repeat the same LOCs, it would be nice if we write a helper that creates `makeRequest` stubs which resolves with the `existingCts`, then we only need to run the `migrationParser`:

```js
const makeRequest = resolveWith(existingCts)

try {
  createMigrationParser(makeRequest)(migrationFn)
} catch (error) {
  expect(error.details).to.eql({ ... })
}
```

Reply:
for now I've extracted these function calls into a `validateMigration` helper